azkaban-common/src/test/java/azkaban/executor/MockExecutorLoader.java

```diff
@@ -54,7 +54,8 @@
   @Override
   public void uploadExecutableFlow(ExecutableFlow flow)
       throws ExecutorManagerException {
-    flows.put(flow.getExecutionId(), flow);
+    ExecutableFlow exFlow = ExecutableFlow.createExecutableFlowFromObject(flow.toObject());
```

Comment:
"the current flow status, which is not synchronized with the job runner threads"
Why isn't it synchronized?